Applications/mnist/Tensorflow/Training_Keras.py

```diff
@@ -0,0 +1,183 @@
+# @bug		No known bugs except for NYI items
+#
+# mnist example
+# inputlayer -> conv2d (5x5) 6 filters -> pooling2d 2x2 (valid) -> conv2d (5x5) 12 filters ->
+#               pooling2d 2x2 (valid) -> flatten -> fully connected layer -> 10 class
+#
+
+import random
+import struct
+import os
+import tensorflow as tf
+import numpy as np
+from tensorflow.keras.layers import Input, Dense, Conv2D, Flatten, AveragePooling2D
+from tensorflow.keras import models, layers, optimizers
+from tensorflow.keras.callbacks import Callback
+
+TOTAL_TRAIN_DATA_SIZE=100
+TOTAL_LABEL_SIZE=10
+TOTAL_VAL_DATA_SIZE=100
+FEATURE_SIZE=784
+batch_size =32
+Learning = True
+Test = False
+num_epoch = 1500
+
+##
+# @brief Callback to get the data from model during trining
+# @param[in] Callback Keras Callback object
+class History_LAW(Callback):
+    def on_train_begin(self, logs={}):
+        self.epoch=[]
+        self.weights =[]
+        self.history ={}
+        self.weights.append(self.model.layers[0].get_weights())
+    
+    def on_epoch_end(self, epoch, logs={}):
+        self.weights.append(self.model.layers[0].get_weights())
+
+##
+# @brief input data generater with batch_size
+#        ( 32 x 28 x 28 x 1 :training & 32 x 10 : labels )
+# @param[in] x_data : total input data
+# @param[in] y_data : total label data
+# @param[in] batch_size : batch_size
+# @return (x_batch, y_batch)
+def datagen( x_data, y_data, batch_size):
+    size=len(x_data)
+    while True:
+        idx = np.random.permutation(size)
+        x_data=x_data[idx]
+        y_data=y_data[idx]
+
+        for i in range(size // batch_size):
+            x_batch = x_data[i*batch_size: (i+1)*batch_size]
+            x_batch=np.reshape(x_batch, (batch_size, 1, 28,28))
+            x_batch=np.transpose(x_batch, [0,2,3,1])
+            y_batch = y_data[i*batch_size: (i+1)*batch_size]
+            
+
+            yield x_batch, y_batch
+            
+##
+# @brief load input data from file
+# @return (InputVector, InputLabel, Validation Vector, ValidationLabel)
+def load_data():
+    tf.compat.v1.set_random_seed(777)
+
+    data_size = TOTAL_TRAIN_DATA_SIZE;
+    InputVector = np.zeros((TOTAL_LABEL_SIZE*data_size, FEATURE_SIZE),dtype=np.float32)
+    InputLabel = np.zeros((TOTAL_LABEL_SIZE*data_size, TOTAL_LABEL_SIZE),dtype=np.float32)
+
+    ValVector = np.zeros((TOTAL_LABEL_SIZE*TOTAL_VAL_DATA_SIZE,FEATURE_SIZE),dtype=np.float32)
+    ValLabel = np.zeros((TOTAL_LABEL_SIZE*TOTAL_VAL_DATA_SIZE, TOTAL_LABEL_SIZE),dtype=np.float32)
+
+    #read Input & Label    
+
+    fin = open('mnist_trainingSet.dat','rb')
+    for i in range(TOTAL_LABEL_SIZE*TOTAL_VAL_DATA_SIZE):
+        for j in range(FEATURE_SIZE):
+            data_str = fin.read(4)
+            ValVector[i,j] = struct.unpack('f',data_str)[0]
+        for j in range(TOTAL_LABEL_SIZE):
+            data_str = fin.read(4)
+            ValLabel[i,j] = struct.unpack('f',data_str)[0]
+    fin.close()
+
+    fin=open('mnist_trainingSet.dat','rb')
```

Comment:
so in this case, input and validation are same?
if yes, please add a comment here on this. Otherwise this might look like a bug later.

Reply:
done. Thanks